### PR TITLE
When Deleting Repository only explicitly close PRs whose base is not this repository

### LINKED ISSUE #14775: Delete repository don't send issues notifications
When deleting a repository, I received many emails about closing the issues of the repository.

### PR DESCRIPTION
**Pull request by zeripath**
Fix #14775

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: services/pull/pull.go [hunk 1]
```diff
@@ -499,6 +499,9 @@ func CloseRepoBranchesPulls(doer *models.User, repo *models.Repository) error {
 		}
 
 		for _, pr := range prs {
+			if pr.BaseRepoID == repo.ID {
```
Comment (noerw) [inline services/pull/pull.go:504]: Can you add a comment describing which case this covers, it took me a moment to grasp
I.e. `We want to close close PRs targeting this repo, but avoid closing PRs originating from this repo`.

**Review by noerw — APPROVED** [review]

**Comment by noerw** [comment]
closes #14777

File: services/pull/pull.go (hunk 1 above)
Comment (zeripath) [reply services/pull/pull.go:504]:
```suggestion
			// If the base repository for this pr is this repository there is no need to close it
			// as it is going to be deleted anyway
			if pr.BaseRepoID == repo.ID {
```

File: services/pull/pull.go [hunk 2]
```diff
@@ -480,7 +480,7 @@ func CloseBranchPulls(doer *models.User, repoID int64, branch string) error {
 	return nil
 }
 
-// CloseRepoBranchesPulls close all pull requests which head branches are in the given repository
+// CloseRepoBranchesPulls close all pull requests which head branches are in the given repository, but whose base repo is not in the given repository
 func CloseRepoBranchesPulls(doer *models.User, repo *models.Repository) error {
```
Comment (6543) [inline services/pull/pull.go:484]: hmm I think the this is a help func for delete repo.

And so ... In my opinion I would move it into services/repository + unexport it.

File: services/pull/pull.go [hunk 3]
```diff
@@ -480,7 +480,7 @@ func CloseBranchPulls(doer *models.User, repoID int64, branch string) error {
 	return nil
 }
 
-// CloseRepoBranchesPulls close all pull requests which head branches are in the given repository
+// CloseRepoBranchesPulls close all pull requests which head branches are in the given repository, but whose base repo is not in the given repository
```
Comment (6543) [inline services/pull/pull.go]:
```suggestion
// CloseRepoBranchesPulls close all pull requests which head branches are in the given repository, but only whose base repo is not in the given repository
```

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**